plasma_cash/utils/merkle/sparse_merkle_tree.py

```diff
@@ -46,8 +47,30 @@ def create_tree(self, ordered_leaves, depth, default_nodes):
                     else:
                         next_level[index // 2] = sha3(default_nodes[level] + value)
                 prev_index = index
+            tree.append(tree_level)
             tree_level = next_level
         return tree_level[0]
 
+    def create_merkle_proof(self, uid):
+        index = uid
+        proof = b''
+        for level in range(self.depth - 1):
+            sibling_index = index + 1 if index % 2 == 0 else index - 1
+            index = index // 2
+            if sibling_index in self.tree[level]:
+                proof += self.tree[level][sibling_index]
+            else:
+                proof += self.default_nodes[level]
+        return proof
+
+    def is_valid_proof(self, leaf, uid, proof):
+        index = uid
+        hash = leaf
```

Comment:
looks like hash is a word used by python as well, probably better to avoid this word